tests/common-tests.h

```diff
@@ -5735,6 +5754,307 @@ TEST_CASE_METHOD(common_tests, "Into XML vector with several fetches", "[core][x
+
+            std::uint8_t buf[20];
+            std::size_t bytes_read = read_blob.read_from_start(reinterpret_cast<char *>(buf), sizeof(buf));
+
+            CHECK(bytes_read == sizeof(binary_data));
+            for (std::size_t i = 0; i < sizeof(binary_data); ++i) {
+                CHECK(buf[i] == binary_data[i]);
+            }
+        }
+        SECTION("Rowset Blob recognition")
+        {
+            soci::blob blob(sql);
+
+            // Write and retrieve blob from/into database
+            int id = 1;
+            sql << "insert into soci_test (id, b) values(:id, :b)", soci::use(id), soci::use(blob);
+
+            soci::rowset< soci::row > rowSet = sql.prepare << "select id, b from soci_test";
+            bool containedData = false;
+            for (auto it = rowSet.begin(); it != rowSet.end(); ++it) {
+                containedData = true;
+                const soci::row &currentRow = *it;
+
+                soci::data_type type = currentRow.get_properties(0).get_data_type();
+                soci::data_type expectedType = tc_.get_backend() != Backend::Oracle ? soci::dt_integer : soci::dt_long_long;
+                CHECK(type == expectedType);
+
+                CHECK(currentRow.get_properties(1).get_data_type() == soci::dt_blob);
+            }
+            CHECK(containedData);
+        }
+        SECTION("Blob binding")
+        {
+            // Add data
+            soci::blob blob(sql);
+            static_assert(10 <= sizeof(dummy_data), "Underlying assumption violated");
+            blob.write_from_start(dummy_data, 10);
+            const int id = 42;
+            sql << "insert into soci_test (id, b) values(:id, :b)", soci::use(id), soci::use(blob);
+
+            SECTION("into")
+            {
+                soci::blob intoBlob(sql);
+                std::string intoString;
+
+                sql << "select b from soci_test where id=:id", soci::use(id), soci::into(intoBlob);
+                sql << "select b from soci_test where id=:id", soci::use(id), soci::into(intoString);
+
+                char buffer[20];
+                std::size_t written = intoBlob.read_from_start(buffer, sizeof(buffer));
+                CHECK(written == 10);
+                for (std::size_t i = 0; i < 10; ++i) {
+                    CHECK(buffer[i] == dummy_data[i]);
+                    CHECK(intoString[i] == dummy_data[i]);
+                }
+            }
+            SECTION("get")
+            {
+                soci::rowset< soci::row > rowSet = (sql.prepare << "select b from soci_test where id=:id", soci::use(id));
+                bool containedData = false;
+                for (auto it = rowSet.begin(); it != rowSet.end(); ++it) {
+                    containedData = true;
+                    const soci::row &currentRow = *it;
+
+                    // TODO: Figure out how to obtain BLOB contents into a std::string directly
+                    //std::string intoString = currentRow.get<std::string>(0);
```

Comment:
@vadz is there already machinery available that allows specific actions to be applied in order to transform the values stored inside a `row` object into the type requested by the user?
I thought `type_conversion` was the one, but that doesn't seem to work as this can only be specialized on the user-provided type without regard for what the storage type is. Thus the latter is inferred from the specialization, which is not what I want (I don't want to change the storage type for BLOBs).

Reply:
Okay ^^